provisioning_templates/PXELinux/kickstart_default_pxelinux.erb

```diff
@@ -25,20 +25,27 @@ oses:
   #  both current and legacy syntax provided
   options.push("kssendmac", "ks.sendmac", "inst.ks.sendmac")
 
-  # handle non-DHCP environments (e.g. bootdisk)
-  subnet = @host.provision_interface.subnet
-  unless subnet.dhcp_boot_mode?
-    # static network configuration
+  # handle non-DHCP IPv4 or IPv6 environments (e.g. bootdisk)
+  if @host.provision_interface.subnet
+    subnet = @host.provision_interface.subnet
     ip = @host.provision_interface.ip
-    mask = subnet.mask
+    mask_or_cidr = subnet.mask
     gw = subnet.gateway
+  elsif @host.provision_interface.subnet6
+    subnet = @host.provision_interface.subnet6
+    ip = "[#{@host.provision_interface.ip6}]"
+    mask_or_cidr = subnet.cidr
+    gw = subnet.gateway.blank? ? '' : "[#{subnet.gateway}]"
+  end
+  dns_servers = subnet.dns_servers
```

Comment:
Is the interface guaranteed to have either a v4 or v6 interface? There is no else case so subnet may be nil.

Reply:
Fixed, good catch.

Reply:
This is still relevant. You can also move this line in the `if subnet && !subnet.dhcp_boot_mode?` or always use `subnet.dns_servers` instead of `dns_servers`.

Reply:
Yeah now I get it, gonna move it the line.